src/goto-programs/goto_contractor.h

```diff
@@ -337,34 +401,47 @@ class goto_contractort : public goto_functions_algorithm
    * Fourth, inserting assumes in the program to reflect the contracted intervals.
    * @param _goto_functions
    */
-  goto_contractort(goto_functionst &_goto_functions)
+  goto_contractort(
+    goto_functionst &_goto_functions,
+    const namespacet &ns,
+    optionst options)
     : goto_functions_algorithm(true), goto_functions(_goto_functions)
   {
-    initialize_main_function_loops();
-    if(!function_loops.empty())
+    if(options.get_bool_option("goto-contractor-condition"))
     {
       vars = new ibex::Variable(CspMap::MAX_VAR);
-      log_debug(
-        "contractor", "1/4 - Parsing asserts to create CSP Constraints.");
-      get_contractors(_goto_functions);
-      if(contractors.is_empty())
+      goto_contractor_condition(ns, options);
+    }
+    else
+    {
+      initialize_main_function_loops();
+      if(!function_loops.empty())
       {
-        log_status(
-          "Contractors: expression not supported, No Contractors were "
-          "created.");
-        return;
+        vars = new ibex::Variable(CspMap::MAX_VAR);
+        parser = expr_to_ibex_parser(&map, vars);
+        log_debug(
+          "contractor", "1/4 - Parsing asserts to create CSP Constraints.");
+        get_contractors(_goto_functions);
+        if(contractors.is_empty())
+        {
+          log_status(
+            "contractor",
```

Comment:
The additional first "module" string is only needed for log_debug(). This call here will just print "contractor", which is unlikely to be intended.

Reply:
That's actually meant to be log_debug. fixed.